src/node_internals.h

```diff
@@ -210,6 +210,83 @@ inline void NODE_SET_EXTERNAL(v8::Handle<v8::ObjectTemplate> target,
                                                          v8::DontDelete));
 }
 
+enum NodeInstanceType { MAIN, WORKER };
+
+class NodeInstanceData {
+  public:
+    NodeInstanceData(NodeInstanceType node_instance_type,
+                     uv_loop_t* event_loop,
+                     int argc,
+                     const char** argv,
+                     int exec_argc,
+                     const char** exec_argv,
+                     bool use_debug_agent_flag) :
+        node_instance_type_(node_instance_type),
+        exit_code_(1),
+        event_loop_(event_loop),
+        argc_(argc),
+        argv_(argv),
+        exec_argc_(exec_argc),
+        exec_argv_(exec_argv),
+        use_debug_agent_flag_(use_debug_agent_flag) {
+    }
+
+    uv_loop_t* event_loop() const {
+      ASSERT_NE(event_loop_, nullptr);
+      return event_loop_;
+    }
+
+    int exit_code() {
+      CHECK(is_main());
+      return exit_code_;
+    }
+
+    void set_exit_code(int exit_code) {
+      CHECK(is_main());
```

Comment:
Well ultimately it will depend on how the Worker API will turn out and whether it's going to need the exit code, for now I just made it more restrictive in case it won't be needed. If it turns out it's going to be needed, it can always be loosened up afterwards.